### ISSUE TITLE
hdevtools maker should not call stack

### ISSUE THREAD

**Issue by kirelagin** [neomake/neomake#1176]
The correct way to use `hdevtools` is to simply run it, without trying to deal explicitly with stack: nowadays `hdevtools` detects `stack.yaml` and does everything itself (hdevtools/hdevtools#6).

I managed to make it work with my stack project by simply removing the `s:TryStack()` call, but I am not sure, what to do with all the other fiddling related to `GHC-PACKAGE-PATH`.

It would be great if @bonds could take a look.

**Comment by bonds** [comment]
I'm able to run hdevtools on both runghc scripts and full cabal-based 
projects without the need for stack or GHC-PACKAGE-PATH shennanigans.  
Feel free to submit a PR ripping all that stuff out.

**Comment by kirelagin** [comment]
@bonds Is it really true that “stack may be able to find a maker binary that's not on the normal path”? Are there really usecases for this? This whole caching business seems to be overly complicated and I doubt that it pays off.

**Comment by bonds** [comment]
Yes. Stack based projects can exist in their own sandbox, with their own 
versions of hdevtools and other makers. You could, in fact, not have 
hdevtools installed at all except in your project, so that $ hdevtools 
errors but $ stack exec hdevtools works.

The caching was added because someone complained about the startup 
time...there's a ticket somewhere for you to look at the discussion.  
Its not a big deal on a fast machine, but if you are on a slower machine 
it is.

**Comment by kirelagin** [comment]
> so that $ hdevtools errors but $ stack exec hdevtools works.

The issue with this is that `hdevtools` _cannot_ be used as `stack exec hdevtools`, it will error out (at least currently and, as far as I can tell, this is not going to be fixed any time soon).

To be honest, I do not see how having _tooling_ per-project can be useful, but, ok, if you think some people might need this, I’ll try to figure something out (keeping in mind that this will _not_ work for `hdevtools` in any case).

**Comment by bonds** [comment]
Sorry, my memory is crap. My guess is I encountered the need to have use 
stack exec <util> instead of just plain <util> when working on projects 
that used different versions of ghc (project A on ghc 7.8, project B on 
ghc 7.10), and thus, different and incompatible versions of ghc-mod and 
the like. stack exec ghc-mod works with both project A and B in that 
scenario, plain ghc-mod will use the ghc-mod on your path, which may 
work for one but not the other.

That is the point of stack exec. Its good that hdevtools doesn't need it 
anymore, but I don't think that's the case for the other makers.
